xclim/testing/tests/test_run_length.py

```diff
@@ -558,7 +558,7 @@ def test_run_with_dates_different_calendars(self, calendar, expected):
             .resample(time="AS-MAR")
             .map(rl.last_run_before_date, date="03-02", window=2)
         )
-        np.testing.assert_array_equal(out.values[1:], np.array(expected) + 1)
+        np.testing.assert_array_equal(out.values[1:], np.array(expected))
```

Comment:
The previous output of `last_run_before_date` was not what I would expect, and the modified implementation I have gives something else. So I modified the implementation of the test, but let me explain and we can decide.

The example has a run starting on march 1, which means doy=[61,60] on a [leap year, non-leap year]. The call to `first_run_after_"03-01"`  and `last_run_before_"03-02"`^[both boundary dates are included given how these functions are constructed] are meant to isolate this run. 
        - `First_run_after "03-01"` gives [61,60], which is the `expected` array
        - Now, I find that intuitively, `Last_run_before_"03-02"` should give the same answer. This was not like this before, because this output was compared to  `expected+1` instead.
        - With my current version of `Last run before "03-02"`, I also get [61,60], which is I think 
        what we should expect from the function last run. For a *same run* selected by `first_run_after_date` and  `last_run_before_date`, I think it would make sense to get the same answer.  Of course, there might be multiple runs possibles, so that the two algorithm will give different output.

Reply:
... should have read the doc...
`last_run` : `Return the index of the **last item** of the last run of at least a given length.`
The function does what it claims to do. I could include an argument `index` ("first" or "last") in `_boundary_run` that lets the choice of first or last element.

Reply:
That would be good, IMO. Breaking changes are fine so long as we can replicate past behaviour and results.

Reply:
In order to be less confusing, you could try going with `position` or `attribute/value_position`, maybe?

I agree that the level of control is starting to make it a bit confusing.

Reply:
That already sounds better!